pkgs/development/libraries/libxsmm/default.nix

```
@@ -35,10 +35,23 @@ stdenv.mkDerivation rec {
     "STATIC=${static}"
   ];
 
-  postInstall = ''
+  postInstall =
+  let rpath = lib.makeLibraryPath [
+        stdenv.cc.libc
+        stdenv.cc.cc.lib
+        gfortran.cc.lib
+        "$out"
+      ];
+  in ''
     mkdir -p $dev/lib/pkgconfig
     mv $out/lib/*.pc $dev/lib/pkgconfig
 
+    # Set rpath for all binaries and shared libraries explicitly.
+    # Otherwise the rpath contains /build/source/lib paths
+    for F in $out/bin/* $out/lib/*.so*; do
+      patchelf --set-rpath ${rpath} $F
+    done
+
     moveToOutput "share/libxsmm" "$doc"
```

```suggestion
    moveToOutput "share/libxsmm" '${!outputDoc}"
```
(is the proper method, as I recently learned myself)